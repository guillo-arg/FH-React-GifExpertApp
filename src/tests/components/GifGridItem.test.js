import {shallow} from 'enzyme';
import { GifGridItem } from "../../components/GifGridItem";

describe('Test sobre el componentes <GifGridItem />', () => {
  
    const title = 'Un título';
    const url = 'http://localhost';
    const wrapper  = shallow(<GifGridItem title={title} url={url} />);

    test('Debe mostrar el componente correctamente', () => {
      
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe tener un párrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);

    });
    
    test('Debe tener la imagen igual al url y alt de los props', () => {
       const img = wrapper.find('img');
    //    console.log(img.props().src);
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('el div debe tener la clase animate__fadeIn', () => {
        const div = wrapper.find('div');
        const clase = 'animate__fadeIn';

        // console.log(div.prop('className'));
        expect(div.prop('className').includes(clase)).toBe(true);


    });
    
    
    
});
