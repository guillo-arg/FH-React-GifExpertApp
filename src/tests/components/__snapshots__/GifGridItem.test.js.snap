// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test sobre el componentes <GifGridItem /> Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="http://localhost"
  />
  <p>
    Un título
  </p>
</div>
`;
